projects/ADuCM3029_demo_cn0397/src/cn0397_app.cpp

```diff
@@ -0,0 +1,519 @@
+
+    /* WHILE(forever) */
+    while(1u) {
+
+        /* Dispatch events for two seconds - they will arrive in the application callback */
+        eResult = adi_ble_DispatchEvents(ADI_APP_DISPATCH_TIMEOUT);
+        DEBUG_RESULT("Error dispatching events to the callback.\r\n", eResult, ADI_BLER_SUCCESS);
+
+        /* If connected, send data */
+        if(gbConnected == true) {
+            /* Get the connection info. The connection handle will be used to identify which remote device to send data to. */
+            adi_ble_GetConnectionInfo(&sConnInfo);
+
+            if(reg_flag == 0)	{
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(reg_pkt),(uint8_t*)&reg_pkt);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt0),(uint8_t*)&name_pkt0);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt1),(uint8_t*)&name_pkt1);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt2),(uint8_t*)&name_pkt2);
+            }
+
+            /* Get the red, green and blue visual light data */
+            pLight->getLightIntensity((float *)&data_pkt.Sensor_Data1);
+
+            if(reg_flag == 1)	{
+            	/* Send the data packet to the remote device */
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            									sizeof(data_pkt),(uint8_t*)&data_pkt);
+            	 DEBUG_MESSAGE("Error sending the data.\r\n");
+            }
+        }
+        /* If disconnected, switch to advertising mode */
+        else
+        {
+            if (geMode != PERIPHERAL_ADV_MODE)
+            {
+                SetAdvertisingMode();
+            }
+        }
+
+    } /* ENDWHILE */
+}
+#else
+/*!
+ * @brief      Standalone visible light demo.
+ *
+ * @details    Standalone mode just prints the red, green and blue values to the console
+ *			   in debug mode and the terminal in release mode.
+ */
+void VisibleLightStandaloneMode(VisibleLight *pLight)
+{
+    /* WHILE(forever) */
+    while(1u) {
+        ADI_VISUAL_LIGHT_DATA eIntensity;
+        ADI_VISUAL_LIGHT_DATA eConcentration;
+
+        /* Get the red, green and blue visual light data and concentration. */
+        pLight->getLightIntensity((float *)&eIntensity.fData_Red);
+        pLight->getLightConcentration((float *)&eConcentration.fData_Red);
+
+#ifndef ADI_DEBUG
+        uint8_t i;
+
+        /* Get the concentraion bar. This is used to draw a color line on the terminal or console. */
+        CN0397_SetBar(eConcentration.fData_Red, &BarLine[0]);
+        CN0397_SetBar(eConcentration.fData_Green, &BarLine[1]);
+        CN0397_SetBar(eConcentration.fData_Blue, &BarLine[2]);
+
+        /* Print the Red concentration bar */
+        DEBUG_MESSAGE("\t%s channel:[", "RED");
+
+        for(i = 0; i < BarLine[0] ; i++) {
+            DEBUG_MESSAGE("\033[2;%dm%c\033[0m", 31, 219);
```

Comment:
Not sure, but I think DEBUG_MESSAGE is not defined in Release mode. We should be careful when implementing logic using it because it might not work in Release.